OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -159,6 +159,9 @@ public void Resurrect(Actor self, Actor repairer)
 
 		public void InflictDamage(Actor self, Actor attacker, Damage damage, bool ignoreModifiers)
 		{
+			if (self.Owner.PlayerActor.Trait<DeveloperMode>().Immortal)
```

Comment:
`InflictDamage` is a hot code path, we really don't want to be doing trait lookups here!

Reply:
good point. Should I try to make the code less costly (simple variable lookup) or do we need to avoid the path completely?

Reply:
See #20282 and before that #20277.

Reply:
Thanks!